src/transformers/generation_utils_dist_process.py

```diff
@@ -0,0 +1,249 @@
+# coding=utf-8
+# Copyright 2020 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from typing import Iterable, List
+
+import torch
+from torch.nn import functional as F
+
+
+class DistProcessorList(list):
```

Comment:
I think I would also avoid the use of the abreviation `dist` which is often used for "distributed".

You can use `distributions` or `probability`